### ISSUE TITLE
Connection is already closed.

### ISSUE THREAD

**Issue by fjramirez1987** [peterbuga/HASS-sonoff-ewelink#126]
I have problems when executing a scene with some sonoff devices. It seems that the connection is closed when several consecutive requests are made.

> websocket error: Connection is already closed.

**Comment by peterbuga** [comment]
duplicate of https://github.com/peterbuga/HASS-sonoff-ewelink/issues/92

TLDR; quick patch try to add a delay of 1-2 seconds between the actions

**Comment by fjramirez1987** [comment]
Is it possible to add that delay in the custom component?  I am using scenes, the scenes do not allow delays.

**Comment by peterbuga** [comment]
you could create an automation-alias entity to turn on/off the devices used in the scenes, add the delays there and use this in the scene. i know it's a bit cumbersome but this is the best it can be done right now.

i have some solutions lined up but no time or mood to test & implement them.
(ps: the eWeLink app does the same thing when activating a scene, changes the state of a device 1 by 1 not all at once)

**Comment by mattsaxon** [comment]
All, just in case you might want to use an alternate mechanism, the LAN based Sonoff components discussed here work very well in parallel, I for example have over 20 devices controlled this way. https://community.home-assistant.io/t/new-custom-component-sonoff-lan-mode-local-with-stock-firmware/88132/27

**Comment by fjramirez1987** [comment]
I had that same idea but with a script.  I will try to look at the component code to try to do something.  I am a programmer, but HA is new to me.  Thanks for your time.

**Comment by peterbuga** [comment]
for the time being i'm going to keep the component online-only & autodiscovery(-ish) as you might soon find out that lan-mode is quite unreliable (works for some devices doesn't for others) and it does require some manual setup (and this raises a lot of *doesn't-work-for-me* issues)
you guys might find it trivial but if you browse through the issues around here there are persons at a level "unable to find the `custom_components` folder" so there's that...

**Comment by mattsaxon** [comment]
@peterbuga, agreed, its more difficult to setup, but once done so, the reliability is excellent. If someone is about to embark on some custom coding, I think the LAN mode components are a possible option and can be used in parallel with your component too. @fjramirez1987, if you'd like some assistant, I'm happy to help out with getting you started. Please post issues to my repo here https://github.com/mattsaxon/sonoff-lan-mode-homeassistant

**Comment by fjramirez1987** [comment]
true @peterbuga, I have taken a look at the component that says @mattsaxon and it seems unclear to me.  I prefer Peter's component, simple and functional.  Thank you @mattsaxon
